Release workflows

kgangineni opened the pull request:
### Summary of changes
  - Modified release.yml to trigger automatically on merge to main branch
  - Removed manual workflow dispatch trigger and version input parameter
  - Simplified workflow to always use the version from codebase
  - Adds workflow to create release PR
  - Example release [PR](https://github.com/paypal/paypal-android/pull/356)

 ### Checklist

 - [ ] Added a changelog entry

### Authors
> List GitHub usernames for everyone who contributed to this pull request.

-

kgangineni committed:
trigger snapshot workflow on merge to develop

kgangineni committed:
- Modified release.yml to trigger automatically on merge to main branch
  - Removed manual workflow dispatch trigger and version input parameter
  - Added version extraction from build.gradle to use in release process
  - Simplified workflow to always use the version from codebase
  - Updated CHANGELOG handling to add date automatically
  - Streamlined tag and release creation process

kgangineni committed:
⏺ Add workflow to create release PRs

  - Added GitHub Actions workflow for creating release PRs with version input
  - Automated version updates in build.gradle and CHANGELOG.md
  - Created standardized PR with release branch targeting main
  - Integrated with existing Gradle version management tasks

kgangineni committed:
rename release workflow name to Release On Merge To Main

kgangineni committed:
Remove automatic snapshot release after main release

kgangineni committed:
Merge branch 'develop' into release_workflows

kgangineni committed:
Add sync PR from main to develop workflow after release

kgangineni committed:
- logging to release workflow and related actions
- added github actor who is rleasing

sshropshire commented on .github/workflows/create_release_pr.yml:
```diff
@@ -0,0 +1,57 @@
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "Prepare release ${{ github.event.inputs.versionName }}"
+          branch: release/${{ github.event.inputs.versionName }}
+          delete-branch: false
```
```suggestion
          delete-branch: true
```
Would it make sense to delete the branch afterward? We may end up with a bunch of dangling refs. We do subsequently tag the branch before release too, so we shouldn't need the branch after it's been merged.

sshropshire commented on .github/workflows/release.yml:
```diff
@@ -149,30 +165,26 @@ jobs:
+          git push origin $VERSION
       - name: Save changelog entries to a file
-        run: sed -e '1,/##/d' -e '/##/,$d' CHANGELOG.md > changelog_entries.md
+        run: sed -e '1,/##/d'e '/##/,$d' CHANGELOG.md > changelog_entries.md
```
Is the `-e` supposed to be removed here?

sshropshire commented on .github/workflows/release.yml:
```diff
@@ -149,30 +165,26 @@ jobs:
+          
+          # CHANGELOG is already updated by the release PR process
+          # Just create the tag without modifying files
+          git tag $VERSION -a -m "Release $VERSION, automated from main branch"
```
Can we add back the `${{ github.actor }}` part? I feel like that's good information for the tag.

kgangineni committed:
revert logging

kgangineni committed:
PR comments
- include github_actor in release
- add `-e` in change log generation, restoring previous state

kgangineni committed:
PR comments
- rename sync/* branch to sync-*
